src/pools/dynamic-pool.ts

```diff
@@ -0,0 +1,20 @@
+import type { JSONValue } from '../utility-types'
+import type { AbstractPool, IWorker } from './abstract-pool'
+
+/**
+ * Contract definition for a dynamic poolifier pool.
+ *
+ * @template Worker Type of worker which manages this pool.
+ * @template Data Type of data sent to the worker.
+ * @template Response Type of response of execution.
+ */
+export interface IDynamicPool<
+  Worker extends IWorker,
+  Data extends JSONValue = JSONValue,
+  Response extends JSONValue = JSONValue
+> extends AbstractPool<Worker, Data, Response> {
+  /**
+   * Maximum number of workers that can be created by this pool.
+   */
+  readonly max: number
+}
```

Comment:
Are you sure to still thinking a placeholder for types definition is not necessary? :)

Reply:
I'm not sure if I understood you correctly :confused:  
Sorry if something was lost in translation

I created this interface so Dynamic Pools are forced to implement the `max` property
The member functions still cannot be defined by an interface, cause they are `protected`
Functions forced by an interface must be `public` by design and we don't want that specifically by our design.

Basically I write TypeScript types/interfaces to represent the structure and real estimates of the project. They are like contract definitions to shape the API and/or function in- and out-puts.

Reply:
At some point, usually code starts to gather of all them in the same place. It's just nitpicking on code modularity.